src/main/java/com/amazon/opendistroforelasticsearch/security/DefaultObjectMapper.java

```diff
@@ -214,6 +217,19 @@ public T run() throws Exception {
             return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
                 @Override
                 public T run() throws Exception {
+                    if (jt.getBindings().getTypeParameters().get(0).getRawClass() == InternalUserV6.class) {
+                        JsonNode root = readTree(string);
+                        Iterator<JsonNode> internalUsers = root.elements();
+                        while (internalUsers.hasNext()) {
+                            ObjectNode internalUser = (ObjectNode) internalUsers.next();
+                            JsonNode plainTextPassword = internalUser.get("password");
+                            if (plainTextPassword == null)
+                                continue;
+                            if (plainTextPassword.asText().length() == 0)
+                                internalUser.remove("password");
+                        }
+                        return objectMapper.readValue(writeValueAsString(root, false), jt);
```

Comment:
Can you confirm if we are using default object mapper and not non-validating object mapper ?

Reply:
Migration hits DefaultObjectMapper.java based on my debugging. It calls com.fasterxml.jackson.databind.ObjectMapper.java